data/mods/gen1/scripts.js

```diff
@@ -30,9 +30,10 @@ let BattleScripts = {
 		getStat(statName, unmodified) {
 			statName = toId(statName);
 			if (statName === 'hp') return this.maxhp;
-			if (unmodified) return this.storedStats[statName];
+			const s = /** @type {StatNameExceptHP} */(statName);
```

Comment:
Instead of casting, `getStat` should outright throw if receiving `statName === 'hp'`, and the function signature should take a `StatNameExceptHP`.

Reply:
Done.